reana_server/utils.py

```diff
@@ -149,3 +154,42 @@ def _get_user_from_invenio_user(id):
     if not user:
         raise ValueError('No users registered with this id')
     return user
+
+
+def _get_yaml_file_from_gitlab(webhook_data, user_id):
+    gitlab_api = GITLAB_URL + "api/v4/projects/{0}" + \
+                 "/repository/files/{1}/raw?ref={2}&access_token={3}"
+    reana_yaml = 'reana.yaml'
+    if webhook_data['object_kind'] == 'push':
+        branch = webhook_data['project']['default_branch']
+        commit_sha = webhook_data['checkout_sha']
+    elif webhook_data['object_kind'] == 'merge_request':
+        branch = webhook_data['object_attributes']['source_branch']
+        commit_sha = webhook_data['object_attributes']['last_commit']['id']
+    secrets_store = REANAUserSecretsStore(str(user_id))
+    gitlab_token = secrets_store.get_secret_value('gitlab_access_token')
+    project_id = webhook_data['project']['id']
+    yaml_file = requests.get(gitlab_api.format(project_id, reana_yaml,
+                                               branch, gitlab_token))
+    return yaml.load(yaml_file.content), \
+        webhook_data['project']['path_with_namespace'], branch, \
+        commit_sha
+
+
+def _parse_gitlab_secret(gitlab_response):
```

Comment:
This is more formatting than parsing right? I would rename to `_format_gitlab_secrets` or similar.